src/PatchOrchestrationApplication/NodeAgentNTService/src/Manager/WindowsUpdateManager.cs

```diff
@@ -324,14 +328,22 @@ private bool HandleWUOperationStates(CancellationToken cancellationToken)
                 case NodeAgentSfUtilityExitCodes.DownloadCompleted:
                 case NodeAgentSfUtilityExitCodes.InstallationApproved:
                     {
+                        if(wuOperationState == NodeAgentSfUtilityExitCodes.DownloadCompleted)
+                        {
+                            string WUDownloadComplete = string.Format("Windows updates downloaded, waiting for installation approval from Repair Manager.");
+                            this._nodeAgentSfUtility.ReportWUStatusUpdateOnCoordinatorService(WUOperationStatusUpdate, WUDownloadComplete, HealthState.Ok,-1, TimeSpan.FromMinutes(this._serviceSettings.OperationTimeOutInMinutes));
+                        }
                         NodeAgentSfUtilityExitCodes exitCodes = this.WaitForInstallationApproval(cancellationToken);
                         if (exitCodes.Equals(NodeAgentSfUtilityExitCodes.Failure))
                         {
                             _eventSource.ErrorMessage("Not able to move from DownloadCompleted state to InstallationApproved state.");
                             reschedule = true;
                             break;
                         }
-                        
+
+                        string wUStatusUpdate = string.Format("Windows update installation in progress.");
+                        this._nodeAgentSfUtility.ReportWUStatusUpdateOnCoordinatorService(WUOperationStatusUpdate, wUStatusUpdate, HealthState.Ok,-1, TimeSpan.FromMinutes(this._serviceSettings.OperationTimeOutInMinutes));
+
```

Comment:
Also, I see that we have updates for success scenarios. What about failure cases?

Reply:
We dont post success or failure. We only say that we have attempted.

Reply:
I meant, are there some failure cases where we need to post explicit health?

---
In reply to: [281163224](https://github.com/microsoft/Service-Fabric-POA/pull/14#discussion_r281163224) [](ancestors = 281163224)